service: Wait on NetworkManager-wait-online.service

cathay4t opened the pull request:
When applying this policy via nmstate.service (`/etc/nmstate/desired.yml`):

```yml
capture:
  default-gw-route: routes.running.destination=="0.0.0.0/0"
  default-gw-iface: interfaces.name==capture.default-gw-route.routes.running.0.next-hop-interface
desired:
  interfaces:
  - name: "{{ capture.default-gw-iface.interfaces.0.name }}"
    type: ethernet
    state: up
  - name: br-ex
    type: linux-bridge
    copy-mac-from: "{{ capture.default-gw-iface.interfaces.0.name }}"
    state: up
    ipv4: "{{ capture.default-gw-iface.interfaces.0.ipv4 }}"
    ipv6: "{{ capture.default-gw-iface.interfaces.0.ipv6 }}"
    bridge:
      port:
      - name: "{{ capture.default-gw-iface.interfaces.0.name }}"
```

We got failure indicate no interface is holding route to `0.0.0.0/0`.

This is because `nmstate.service` did not wait NetworkManager activating
all connections.

Hence change `nmstate.service` to wait on
`NetworkManager-wait-online.service`.

No test case created because this require OS boot up.

Manually tested.

bengal commented:
I think this will not work if NetworkManager-wait-online.service is disabled. Since nmstate.service depends on it, it should also `Require=` the service. Otherwise it's possible that NetworkManager-wait-online.service is not pulled into the boot transaction?

cathay4t committed:
service: Wait on NetworkManager-wait-online.service

When applying this policy via nmstate.service (`/etc/nmstate/desired.yml`):

```yml
capture:
  default-gw-route: routes.running.destination=="0.0.0.0/0"
  default-gw-iface: interfaces.name==capture.default-gw-route.routes.running.0.next-hop-interface
desired:
  interfaces:
  - name: "{{ capture.default-gw-iface.interfaces.0.name }}"
    type: ethernet
    state: up
  - name: br-ex
    type: linux-bridge
    copy-mac-from: "{{ capture.default-gw-iface.interfaces.0.name }}"
    state: up
    ipv4: "{{ capture.default-gw-iface.interfaces.0.ipv4 }}"
    ipv6: "{{ capture.default-gw-iface.interfaces.0.ipv6 }}"
    bridge:
      port:
      - name: "{{ capture.default-gw-iface.interfaces.0.name }}"
```

We got failure indicate no interface is holding route to `0.0.0.0/0`.

This is because `nmstate.service` did not wait NetworkManager activating
all connections.

Hence change `nmstate.service` to wait on
`NetworkManager-wait-online.service`.

No test case created because this require OS boot up.

Manually tested.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Added `Requires=NetworkManager-wait-online.service`. Thanks.
